### PR TITLE
make emagged marker component (fixed version of #13867)

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
broke last branch hope i didnt do it again

adds `EmaggedComponent` marker to control emagging (prevent emagging a thing twice, unless you really want to)

will eventually replace the ~8 things that use Emagged/IsEmagged/!Safe as component fields

**Changelog**
no cl no fun

### CODE REVIEW

File: Content.Shared/Emag/Systems/EmagSystem.cs [hunk 1]
```diff
@@ -156,12 +156,28 @@ public bool TryUseEmag(EntityUid emag, EntityUid user, EntityUid target, EmagCom
...
+            // prevent emagging twice
+            if (IsEmagged(target))
+                return false;
+
             var emaggedEvent = new GotEmaggedEvent(user);
             RaiseLocalEvent(target, ref emaggedEvent);
+
+            // add marker component unless the handler wants repeatable emagging
+            if (emaggedEvent.Emag)
+                EntityManager.AddComponent<EmaggedComponent>(target);
```
Comment (0x6273) [inline Content.Shared/Emag/Systems/EmagSystem.cs:168]: In systems you should use the EntitySystem.Proxy methods instead.

File: Content.Shared/Emag/Systems/EmagSystem.cs [hunk 2]
```diff
@@ -156,12 +156,28 @@ public bool TryUseEmag(EntityUid emag, EntityUid user, EntityUid target, EmagCom
...
             return emaggedEvent.Handled;
         }
+
+        /// <summary>
+        /// Returns whether an entity has the emagged marker component
+        /// </summary>
+        public bool IsEmagged(EntityUid uid)
+        {
+            return EntityManager.HasComponent<EmaggedComponent>(uid);
+        }
```
Comment (0x6273) [inline Content.Shared/Emag/Systems/EmagSystem.cs:178]: Why is this needed? You can just do HasComp directly or use EntityQuery to check for marker comps.

Comment (deltanedas) [reply Content.Shared/Emag/Systems/EmagSystem.cs:178]: just a little helper i thought i would use it more, can remove

File: Content.Shared/Emag/Systems/EmagSystem.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Shared/Emag/Systems/EmagSystem.cs:168]: wasnt aware of it back when i wrote it, thanks for spotting

File: Content.Shared/Emag/Components/EmaggedComponent.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+namespace Content.Shared.Emag.Components;
+
+/// <summary>
+/// Marker component for emagged entities
+/// </summary>
+[RegisterComponent]
```
Comment (metalgearsloth) [inline Content.Shared/Emag/Components/EmaggedComponent.cs:6]:
```suggestion
[RegisterComponent, NetworkedComponent]
```

File: Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs [hunk 4]
```diff
@@ -91,12 +92,11 @@ private List<string> GetInventory(ReagentDispenserComponent reagentDispenser)
...
         private void OnEmagged(EntityUid uid, ReagentDispenserComponent reagentDispenser, ref GotEmaggedEvent args)
         {
-            if (!reagentDispenser.IsEmagged)
-            {
-                reagentDispenser.IsEmagged = true;
-                args.Handled = true;
-                UpdateUiState(reagentDispenser);
-            }
+            // adding component manually to have correct state
+            args.Emag = false;
```
Comment (metalgearsloth) [inline Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs:96]: Should the event raiser just use EnsureComp<T> instead.

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs (hunk 4 above)
Comment (deltanedas) [reply Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs:96]: didnt know that was a thing, will do it in a bit
